pyvo/discover/image.py

```diff
@@ -0,0 +1,635 @@
+                    l1="t_min", h1="t_max",
+                    l2=self.time_min, h2=self.time_max))
+
+        where_clause = "WHERE "+(" AND ".join(where_parts))
+
+        while self.obscore_recs:
+            rec = self.obscore_recs.pop()
+            try:
+                self._query_one_obscore(rec, where_clause)
+            except Exception as msg:
+                self._log("Obscore {} skipped: {}".format(
+                    rec.res_rec['ivoid'], msg))
+                self.failed_services += 1
+            self.already_queried += 1
+
+    def get_query_stats(self):
+        """returns a tuple of n(total to query), n(already queried)
+        """
+        total_to_query = len(self.obscore_recs)\
+            + len(self.sia1_recs)\
+            + len(self.sia2_recs)
+        return total_to_query, self.already_queried, self.failed_services
+
+    def query_services(self):
+        """queries the discovered image services according to our
+        constraints.
+
+        This creates and fills the results and the log_messages attributes.
+        """
+        if (not self.sia1_recs
+                and not self.sia2_recs
+                and not self.obscore_recs):
+            raise dal.DALQueryError("No services to query.  Unless"
+                " you overrode service selection, you will have to"
+                " loosen your constraints.")
+        self._query_obscore()
+        self._query_sia2()
+        self._query_sia1()
+
+
+def images_globally(
+        *,
+        space: Optional[Tuple[float, float, float]] = None,
+        spectrum: Optional[quantity.Quantity] = None,
+        time: Optional[time.Time] = None,
+        inclusive: bool = False,
+        watcher: Optional[Callable[['ImageDiscoverer', str], None]] = None,
+        timeout: float = 20,
+        services: Optional[registry.RegistryResults] = None)\
+        -> Tuple[List[obscore.ObsCoreMetadata], List[str]]:
+    """returns a collection of ObsCoreMetadata-s matching certain constraints
+    and a list of log lines.
+
+    Parameters
+    ----------
+
+    space :
+        An optional tuple of ra, dec, and the search radius, all in degrees;
+        images returned must intersect a spherical circle described in this
+        way.
+    spectrum :
+        An astropy quantity convertible to a (vacuum) wavelength; images
+        must cover this point in the (electromagnetic) spectrum.
+    time :
+        An astropy time that must be in the observation time of the image
+        (if it declares a time).
+    inclusive :
+        Set to True to incluse services that do not declare their
+        STC coverage.  By 2023, it's a good idea to do that as many
+        relevant archives do not do that.
+    watcher :
+        A callable that will be called with the ImageDiscoverer instance and
+        a string perhaps suitable for displaying to a human.
+    services :
+        An optional `~pyvo.registry.RegistryResults` instance to
```

Comment:
use the public namespace (we may want to clean these up and have `RegistryResults` documented in registry rather than in registry.regtap, I'm sure there are a couple of related issues already)
```suggestion
        An optional `~pyvo.registry.regtap.RegistryResults` instance to
```